apps/google-analytics-4/frontend/src/components/config-screen/map-account-property/MapAccountPropertyDropdown.tsx

```diff
@@ -0,0 +1,94 @@
+import { useState, useEffect } from 'react';
+import { Stack, Box, Select } from '@contentful/f36-components';
+import { AccountSummariesType, ConfigurationWarningTypes } from 'types';
+import {
+  NO_PROPERTIES,
+  getPropertyDeletedMsg,
+  WarningTypes,
+} from 'components/config-screen/WarningDisplay/constants/warningMessages';
+import WarningDisplay from 'components/config-screen/WarningDisplay/WarningDisplay';
+import Note from 'components/common/Note/Note';
+
+interface Props {
+  onSelectionChange: (e: React.ChangeEvent<HTMLSelectElement>) => void;
+  isPropertyIdInOptions: boolean;
+  selectedPropertyId: string;
+  sortedAccountSummaries: AccountSummariesType[];
+  originalPropertyId: string;
+}
+
+const getIdOnly = (unformattedId: string) => {
+  return unformattedId.split('/')[1];
+};
+
+const MapAccountPropertyDropdown = (props: Props) => {
+  const {
+    onSelectionChange,
+    isPropertyIdInOptions,
+    selectedPropertyId,
+    sortedAccountSummaries,
+    originalPropertyId,
+  } = props;
+
+  const [warningType, setWarningType] = useState<ConfigurationWarningTypes>(WarningTypes.Empty);
+  const [tooltipContent, setTooltipContent] = useState<string>('');
+
+  useEffect(() => {
+    let content = '';
+
+    if (selectedPropertyId === originalPropertyId && !isPropertyIdInOptions) {
+      setWarningType(WarningTypes.Error);
+      content += getPropertyDeletedMsg(getIdOnly(originalPropertyId));
+    }
+
+    if (!content) {
+      setWarningType('');
+    }
+
+    setTooltipContent(content);
+  }, [selectedPropertyId, originalPropertyId, isPropertyIdInOptions]);
+
+  const validateSelection = () => {
```

Comment:
Hm I was a tiny bit confused by this at first but I _think_ it's because this doesn't need to be a function. Shouldn't it just be a variable?

```js
const selectionValue = (selectedPropertyId === originalPropertyId && !isPropertyIdInOptions) ? '' : selectedPropertyid
```

and then update below in the component prop

Reply:
Makes sense, updated this!